src/ripple/nodestore/ShardStoragePaths.md

```diff
@@ -0,0 +1,73 @@
+# Shard Storage Paths
+
+## Overview
+
+The shard database stores validated ledgers in logical groups called shards. As of June 2020, a shard stores 16384 ledgers by default. In order to allow users to store shards on multiple devices, the shard database can be configured with several file system paths. Each path provided should refer to a directory on a distinct filesystem, and no two paths should ever correspond to the same filesystem. Violating this restriction will cause the server to inaccurately estimate the amount of space available for storing shards.
```

Comment:
Please limit lines to 80 characters. Rational: one advantage of markdown is it may be read in raw form. The long lines make it more difficult to read in the terminal and in some editors. Also, long lines make it more difficult to leave review comments localized to the area of the issue.

Reply:
For the sentence:
```
Each path provided should refer to a directory on a distinct filesystem, 
and no two paths should ever correspond to the same filesystem.
```
We should be able to check this condition in code - for linux at least - and either error out or warn if the condition isn't met.

Reply:
> Please limit lines to 80 characters. Rational: one advantage of markdown is it may be read in raw form. The long lines make it more difficult to read in the terminal and in some editors. Also, long lines make it more difficult to leave review comments localized to the area of the issue.

Done

Reply:
> For the sentence:
> 
> ```
> Each path provided should refer to a directory on a distinct filesystem, 
> and no two paths should ever correspond to the same filesystem.
> ```
> 
> We should be able to check this condition in code - for linux at least - and either error out or warn if the condition isn't met.

Added this check for Linux. Errors out during initialization.